src/esp/gfx/Renderer.cpp

```diff
@@ -63,9 +63,33 @@ struct Renderer::Impl {
           DepthShader::Flag::UnprojectExistingDepth);
     }
 
+    RenderTarget::Flags renderTargetFlags_ = {};
+    switch (sensor.specification()->sensorType) {
+      case sensor::SensorType::Color:
+        renderTargetFlags_ |= RenderTarget::Flag::RgbaBuffer;
+        break;
+
+      case sensor::SensorType::Depth:
+        renderTargetFlags_ |= RenderTarget::Flag::DepthTexture;
+        break;
+
+      case sensor::SensorType::Semantic:
+        renderTargetFlags_ |= RenderTarget::Flag::ObjectIdBuffer;
+        break;
+
+      default:
+        // I need this default, since sensor type list is long, and without
+        // default clang-tidy will complain
+        break;
+    }
+    if (flags_ & Flag::NoTextures) {
+      // it means no rgba render buffer
+      renderTargetFlags_ &= ~RenderTarget::Flag::RgbaBuffer;
```

Comment:
@erikwijmans : not sure if you see this if clause here, if textures are not loaded, if would not create a rgbaBuffer. And if the user tries to get the color observation it would fail within the rendering call. 

But I would change it here to make an assertion.

Reply:
Makes sense to fail sooner IMO.  The renderer wasn't looking at the sensor spec beforehand so it wasn't obvious that we could have things fail here, but it should have in hindsight